### PR TITLE
Remove js-to-string and simple-js-parser

### PR DESCRIPTION
**Pull request by kennsippell**
This one might be a hard review -- we can chat on the phone if you'd like to go over it together. 

The issue being address here is #150. The root of the problem is that there is some problematic code for serializing/deserializing objects which have properties which are functions. There are a lot of corner cases where this is failing, which I have not detailed. But I generally object to this approah. The bugs were each very odd, and yielded hard to understand behaviors which I can imagine really frustrating developers building on the CHT.

The reason we need to parse the `target.js` file is because [app settings has a partial-summary of the targets](https://github.com/medic/medic-conf/blob/900eb356f7aaf8fa83fabbb5c469ea24c60c0c62/src/fn/compile-app-settings.js#L69). So the code in `parse-targets` can't be eliminated, but it can now just call `require('targets.js')` instead of writing our own buggy JavaScript lexer/parser. The proposed change here allows us to do less custom compilations and transformations. 

My end goal for this code is detailed in #151 -- to use webpack or browserify to compile the script instead of doing this custom aggregating of files. This is influencing some design decisions here, as I'm taking baby steps toward that goal. For example, `nools-extras` now defines a `module.exports` section to allow information hiding, and to allow for its contents to be unit tested.

### CODE REVIEW

File: src/lib/compile-nools-rules.js [hunk 1]
```diff
@@ -17,17 +15,31 @@ function lint(code) {
...
   const noolsLib = fs.read(`${__dirname}/../nools/lib.js`);
 
   const jsCode = templatedJs.fromString(projectDir, `
     var idx1, idx2, r, target;
     var now = Utils.now();
-    ${supportCode}
-    var targets = ${jsToString(targets)};
-    var tasks = ${tasks};
+    var extras = (function() {
+      var module = {};
```
Comment (dianabarsan) [inline src/lib/compile-nools-rules.js]: I updated standard config to run against this code (module.export-ed all `nools-extras` functions and variables + module.export-ed the arrays in `tasks` and `targets`) and this is where I hit an error in the app:
```
Error getting tasks TypeError: Cannot set property 'MS_IN_DAY' of undefined
    at eval (eval at parseAction (index.js:58), <anonymous>:5:349)
```
When this code is minified below, `module` becomes `i` and `exports` is undefined. I hotpatched it by editing this line to:
```
var module = { exports: {} };
```

Were you thinking of a different way of updating the config where this doesn't happen? I just did what seemed most obvious to me.

File: package.json [hunk 2]
```diff
@@ -56,6 +56,7 @@
     "lodash": "^4.17.11",
     "memdown": "^3.0.0",
     "mocha": "^5.2.0",
+    "rewire": "^4.0.1",
```
[the remark this answers is not in the document]
Comment (dianabarsan) [reply package.json]: Ah, nevermind, I saw you're using __get__ in the other PR's as well.

File: src/lib/compile-nools-rules.js (hunk 1 above)
Comment (kennsippell) [reply src/lib/compile-nools-rules.js]: Interesting.. I'm using es6 for muso and as such, have to disable the minify step. I'll do another test pass against standard now. Your fix seems reasonable.

Comment (kennsippell) [reply src/lib/compile-nools-rules.js]: Ya. Good test. Embarrassing to not have tested the mainline scenario. :P

**Comment by kennsippell** [comment]
@dianabarsan I'm also updating the code which generates the contact summary script. It seems like a logical thing to batch into this change, but I can also split it off into its own issue/PR if you would rather nail this one down.

**Comment by dianabarsan** [comment]
Awesome!

**Pull request merged**